src/main/java/com/github/fppt/jedismock/datastructures/streams/StreamId.java

```diff
@@ -0,0 +1,145 @@
+    private final long firstPart;
+    private final long secondPart;
+
+    public long getFirstPart() {
+        return firstPart;
+    }
+
+    public long getSecondPart() {
+        return secondPart;
+    }
+
+    /* 0-0 ID for PRIVATE API usage */
+    StreamId() {
+        this(0, 0);
+    }
+
+    /* ID from long numbers for PRIVATE API usage */
+    public StreamId(long firstPart, long secondPart) {
+        this.firstPart = firstPart;
+        this.secondPart = secondPart;
+    }
+
+    public StreamId(String key) throws WrongStreamKeyException {
+        String[] parsed = key.split("-");
+
+        switch (parsed.length) {
+            case 2:
+                try {
+                    secondPart = parseUnsignedLong(parsed[1]);
+                } catch (NumberFormatException e) {
+                    throw new WrongStreamKeyException(INVALID_ID_ERROR);
+                }
+
+                try {
+                    firstPart = parseUnsignedLong(parsed[0]);
+                } catch (NumberFormatException e) {
+                    throw new WrongStreamKeyException(INVALID_ID_ERROR);
+                }
+
+                break;
+            case 1:
+                secondPart = 0;
+
+                try {
+                    firstPart = parseUnsignedLong(parsed[0]);
+                } catch (NumberFormatException e) {
+                    throw new WrongStreamKeyException(INVALID_ID_ERROR);
+                }
+
+                break;
+            default:
+                throw new WrongStreamKeyException(INVALID_ID_ERROR);
+        }
+    }
+
+    public StreamId(Slice slice) throws WrongStreamKeyException {
+        this(slice.toString());
+    }
+
+    public StreamId compareToZero() throws WrongStreamKeyException {
+        if (secondPart == 0 && firstPart == 0) {
+            throw new WrongStreamKeyException(ZERO_ERROR);
+        }
+
+        return this;
+    }
+
+    public StreamId increment() throws WrongStreamKeyException {
+        long second = secondPart + 1;
+        long first = firstPart;
+
+        if (compareUnsigned(second, 0) == 0) { // the previous one was 0xFFFFFFFFFFFFFFFF => update the first part
+            if (compareUnsigned(first, -1) == 0) {
+                throw new WrongStreamKeyException(RANGES_START_ID_ERROR);
+            }
+
+            ++first;
+        }
+
+        return new StreamId(first, second);
+    }
+
+    public StreamId decrement() throws WrongStreamKeyException {
+        long second = secondPart - 1;
+        long first = firstPart;
+
+        if (compareUnsigned(second, -1) == 0) { // the previous one was 0x0000000000000000 => update the first part
+            if (compareUnsigned(first, 0) == 0) {
+                throw new WrongStreamKeyException(RANGES_END_ID_ERROR);
+            }
+
+            --first;
+        }
+
+        return new StreamId(first, second);
+    }
+
+    @Override
+    public int compareTo(StreamId other) {
```

Comment:
So if `compareUnsigned != 0` you are comparing it twice. Why not cache the result of the first comparison in a variable?

Why we are bothering with `unsigned` throughout this class at all? Can we consider this values as signed so that we utilize only 1/2 of `long` range? I believe we discussed this earlier on one of our calls.


In particular, this could be rewritten as 

```java
return Comparator.comparing(StreamId::getFirstPart)
                .thenComparing(StreamId::getSecondPart).compare(this, other);
```